disnake/ui/item.py

```diff
@@ -26,56 +37,125 @@
     from typing_extensions import Self
 
     from ..client import Client
-    from ..components import NestedComponent
+    from ..components import ActionRowChildComponent, Component
     from ..enums import ComponentType
     from ..interactions import MessageInteraction
-    from ..types.components import Component as ComponentPayload
+    from ..types.components import ActionRowChildComponent as ActionRowChildComponentPayload
+    from ._types import MediaItemInput
     from .view import View
 
     ItemCallbackType = Callable[[V_co, I, MessageInteraction], Coroutine[Any, Any, Any]]
 
-else:
-    ParamSpec = TypeVar
-
 ClientT = TypeVar("ClientT", bound="Client")
+UIComponentT = TypeVar("UIComponentT", bound="UIComponent")
+
+
+def ensure_ui_component(obj: UIComponentT, name: str) -> UIComponentT:
+    if not isinstance(obj, UIComponent):
+        raise TypeError(f"{name} should be a valid UI component, got {type(obj).__name__}.")
+    return obj
+
 
+def handle_media_item_input(value: MediaItemInput) -> UnfurledMediaItem:
+    if isinstance(value, UnfurledMediaItem):
+        return value
+    elif isinstance(value, str):
+        return UnfurledMediaItem(value)
+    elif isinstance(value, (AssetMixin, Attachment)):
+        return UnfurledMediaItem(value.url)
```

Comment:
Hmm, do you intend to leave `disnake.File` out?
It's not assignable to `MediaItemInput`, nor is it handled here

Reply:
Oh, I should've elaborated on my vision before.
My idea is to allow passing them in, but the user needs to keep a reference to and upload the file as usual themselves
```py
if isinstance(value, disnake.File):
    assert value.filename is not None
    return UnfurledMediaItem(f"attachment://{value.filename}")
```
```py
my_file = File(...)
component = ui.Section(ui.TextDisplay("Woo"), accessory=ui.Thumbnail(my_file))
await inter.response.send_message(components=component, file=my_file, flags=...)
```